### ISSUE TITLE
Read and use maxUniformBufferRange while parsing

### ISSUE THREAD

**Issue by RCalocaO** [KhronosGroup/glslang#1086]
While compiling a shader with this:

> layout(set=0, binding=10, std140) uniform HLSLCC_CBh
> {
> 	vec4 cu_h[5141];
> };
> 

I ran into an issue at runtime since maxUniformBufferRange is smaller:
 Error[Driver:2] Compute info
> ------------
> (0) : error C6020: Constant register limit exceeded at cu_h; more than 1024 registers needed to compiled program
> 

; there should be a way to send those limits (TBuiltInResource doesn't seem to be used) and fail to compile if the shader passes those limits.

**Comment by RCalocaO** [comment]
This is using the 1.0.42.2 glslang.

**Comment by johnkslang** [comment]
What component issued that error? Glslang errors don't look that way, and I don't get an error.  However, without the complete shader (or at least the beginning of it), and knowing the target environment, I can't really reproduce it either.

**Comment by RCalocaO** [comment]
This was a failure inside an Nvidia driver at runtime. I'll try to get a repro case set up (I fixed the shader and it's not straightforward to re-do this on UE4).

**Comment by johnkslang** [comment]
I think this is (likely legitimate) resource size issue with the driver. Closing here.

**Comment by RCalocaO** [comment]
@johnkslang  I agree; however the issue is does glslang use the limits from the config file? If so, it should fail compilation if it goes past those limits.

**Comment by johnkslang** [comment]
Oh, I thought this was outside the scope of covered limits, sorry. I thought the constant limit was different than a buffer range.

Are you sure this isn't about the size of a single constant being too big, versus the buffer?

If you can verify here the spec language that connects the size of that array to a semantically exposed limit, and reopen, I can fix it.

**Comment by dgkoch** [comment]
It's a poor error message, but the size of the UBO would be 5141 * 4 * 4 = 82256 bytes which is greater than the minimum required UBO size of 64K, so at a minimum that should have been caught in glslang.

**Comment by johnkslang** [comment]
Okay, will look at adding that limit test.

**Comment by johnkslang** [comment]
What limit is this? I'm not seeing buffer size limits available in GLSL.

**Comment by dgkoch** [comment]
It would be an API limit, which isn't reflected in the shading language.  
In vulkan this would be `maxUniformBufferRange` for UBOs and `maxStorageBufferRange` for SSBOs. 
In GL this would be `MAX_UNIFORM_BLOCK_SIZE` for UBOs and `MAX_SHADER_STORAGE_BLOCK_SIZE` for SSBOs.
